### DISCUSSION TITLE
Running PrairieLearn using the docker-compose up command

### DISCUSSION THREAD

**Discussion by Prostigus** [PrairieLearn/PrairieLearn#6017] [Q&A] [answered]
Hello, I've been fiddling with the project for a bit and was wondering why the deployment is done with the docker run command instead of the docker-compose up command? I've tried the compose command and have this issue:
![image](https://user-images.githubusercontent.com/71531356/179133821-8a918f6a-4735-41aa-b22a-64a426549359.png)

It seems that the node_modules folder is being overwritten in the container, is there anyway to fix this?

**Comment by nwalters512** [comment, marked as answer]
The `docker-compose.yml` is built for developing PrairieLearn, not necessarily for deploying it or running in production. You'll see that the PrairieLearn repo is actually mounted into `/PrairieLearn`, which is where PL is executed from in the container:

https://github.com/PrairieLearn/PrairieLearn/blob/693ccd8cde2a9353698b083f724ba5df5aa559da/docker-compose.yml#L13

If you remove or comment out that line, that should do the trick.

**Reply by Prostigus** [reply]
Thanks for the reply, I guess I'll have to mount the directories I've changed. 

On another note, in theory, if I was to run this in production, how would I go about adding users to the system?

**Reply by nwalters512** [reply]
If you're trying to run the container with local source, you can always install the Node dependencies on your machine; see https://prairielearn.readthedocs.io/en/latest/installingLocal/ for more details.


Adding users is another question; I'd encourage you to start a new discussion with any relevant details.

**Reply by Prostigus** [reply]
That's a viable option as well, thanks for all the help!
